Dockerfile

```diff
@@ -1,31 +1,84 @@
-FROM alpine:3.10
+# Define the image to feature:
+# --> Operating System: Debian Buster
+# --> Python Version: 3.8.5
+FROM python:3.8.5-buster
 
 # Expecting bind mount at
 ENV PROJECT_DIR=/project/
-# Expect volume mount at
+
+# Expecting volume mount at
 ENV GATORGRADER_DIR=/root/.local/share/
 
+# Set Python version
+ARG PYTHON_VERSION='3.8.5'
+
+# Configure environment variables for Python
+ENV PYTHONUNBUFFERED=1 \
+    # Prevent Python from creating .pyc files
+    PYTHONDONTWRITEBYTECODE=1 \
+    \
+    # Configure pip
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100 \
+    \
+    # Pin the version of Poetry
+    POETRY_VERSION=1.0.10 \
+    # Configure Poetry's installation directory
+    POETRY_HOME="/opt/poetry" \
+    # Make poetry create the virtual environment in the project's root
+    # This virtual environment will be called .venv
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    # Do not ask any interactive questions
+    # during the installation of Poetry
+    POETRY_NO_INTERACTION=1 \
+    \
+    # Specify the paths for using requirements and virtual environments;
+    # this is where the requirements + virtual environment will live
+    PYSETUP_PATH="/opt/pysetup" \
+    VENV_PATH="/opt/pysetup/.venv"
```

Comment:
Is this for project venvs or just poetry's? If it's for projects, we probably want to stick them in `/root/.local/share` so that they persist over container runs (or not if that was designed).

Reply:
I was initially thinking not to persist them in case a student wants to switch between running Programs in the container versus running then regularly on their laptop. What do you think is the best thing to do in this case?

Reply:
Will it cause a greater setup time for the container every run, or will it be fairly minimal? If it's a long time (10 seconds or more would be the max I think), then we should probably persist it.

Reply:
I will go ahead and move this into the /root/.local/share directory. Would that be okay?

Reply:
That sounds good to me.

Reply:
Hi @Michionlion Saejin, I moved it to the suggested directory. With that said, I am not noticing that the Docker container stores anything inside of the specified directory, suggesting that perhaps the variable is not needed. With that said, I am currently inclined to keep the variables in the Dockerfile because I don't notice any adverse affects and because they are mentioned in the GitHub issue tracker as being a part of a "representative Dockerfile".